src/UserDataAccessLayer.cpp

```diff
@@ -0,0 +1,122 @@
+#include "UserData.hpp"
+#include "UserDataAccessLayer.hpp"
+#include <memory>
+#include <string>
+#include <cstdlib>
+#include "Group.hpp"
+#include <utility>
+#include "CommonUtilities.hpp"
+
+namespace
+{
+
+std::optional<Group> inputToGroup(int input)
+{
+	switch(input)
+	{
+		case 1:
+			return std::optional<Group>{Group::weekend};
+		case 2:
+			return std::optional<Group>{Group::evening};
+		default:
+			std::cout << "Incorrect group";
+			postponedConsoleClear(std::chrono::seconds(1));
+			return std::nullopt;
+	}
+}
+
+}
+
+void UserDataAccessLayer::createUser()
+{
+	std::cout << "enter unique nickname: ";
+	std::string nickname;
+	std::cin >> nickname;
+	if(userNickToData.find(nickname) != userNickToData.end())
+	{
+		std::cout << "\n nickname already exists in db! \n";
+		return;
+	}
+
+	std::cout << "\n enter first name: ";
+	std::string name;
+	std::cin >> name;
+
+	std::cout << "\n enter firecode link: ";
+	std::string fcLink;
+	std::cin >> fcLink;
+
+	std::cout << "\n enter github link: ";
+	std::string ghLink;
+	std::cin >> ghLink;
```

Comment:
providing data should be independent from storage. It should be done by some other class (and it's not in the scope of US1)